Description: Reduce optimization level for fs_path.cc to avoid RTL SSA ICE
 The system gcc-12 has a bug in its RTL SSA optimization pass
 (rtl_ssa::sort_accesses) that causes a reproducible internal compiler
 error (Segmentation fault) when compiling fs_path.cc at -O2 in
 virtualized environments. Adding #pragma GCC optimize("O1") to the
 file avoids the buggy pass without affecting any other translation unit.
 Delivered via the gcc-12 quilt system so dpkg-source -b accepts it.
Author: Harkirat S Virk <harkiratsvirk@gmail.com>
Origin: local
Forwarded: no
Last-Update: 2026-05-04
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- /dev/null
+++ b/debian/patches/fix-fspath-gcc-ice.diff
@@ -0,0 +1,8 @@
+# DP: Reduce optimization level for fs_path.cc to avoid RTL SSA ICE
+# DP: in the system GCC when compiling this file during bootstrap.
+
+--- a/src/libstdc++-v3/src/c++17/fs_path.cc
++++ b/src/libstdc++-v3/src/c++17/fs_path.cc
+@@ -1,0 +1,2 @@
++#pragma GCC optimize("O1")
++
--- a/debian/rules.patch
+++ b/debian/rules.patch
@@ -248,5 +248,7 @@
 debian_patches += CVE-2023-4039

+debian_patches += fix-fspath-gcc-ice
+
 # don't remove, this is regularly overwritten, see PR sanitizer/63958.
 #debian_patches += libasan-sparc
